### ISSUE TITLE
Decimals acting strangely when saving bigger precision numbers

### ISSUE THREAD

**Issue by kelostrada** [elixir-ecto/ecto#1952]
### Environment

* Elixir version (elixir -v): `Erlang/OTP 19 [erts-8.2] Elixir 1.4.2`
* Database and version: `Postgres (PostgreSQL) 9.6.1`, `Mysql Ver 14.14 Distrib 5.7.17, for macos10.12 (x86_64)`
* Ecto version (mix deps): `2.1.2` or `2.1.3`
* Database adapter and version (mix deps): `postgrex 0.13.0`, `mariaex 0.8.1`
* Operating system: `macOS Sierra 10.12.3`

### Current behavior

I've created a repo for you to check out (tests): https://github.com/kelostrada/ecto_decimal_bug
You can go check out travis there too, it's simpler to look at what I mean.

Just to explain simply - suppose you have a `User` with a decimal field `balance`. You want to create a new user and input data like this:

    user = Repo.insert!(%User{username: "test", balance: Decimal.new("1.00123")})
    Repo.get(User, user.id)

You would expect that the balance saved to database would be 1.00123. But it saves as 1.00000.

By the way I also increased precision for numerics and it seems it got saved as 1.000000123.

Also the issue seems to be with Ecto, because I checked on postgres and mysql and the issue is the same.

PS. I noticed strange behaviour with NativeDateTime (you can check in tests that date which is saved to MySQL differs from the date which is received from database later - although I expect it's not really a bug, just a problem with saving microseconds in mysql?).

**Comment by josevalim** [comment]
I have fixed the decimal bug on the postgres. Please try master. :)

**Comment by josevalim** [comment]
Btw, thanks for the app reproducing the issue. It makes my life so much easier.

**Comment by josevalim** [comment]
The other issues are also in the Mariaex adapter but I don't have enough knowledge of the codebase to fix them. I will recommend opening an issue there. :) Let me know if the Postgrex fix is good for you so we can ship a new version.

**Comment by kelostrada** [comment]
I've updated postgres to the newest version from master and it seems it is now fixed :) Tests for postgrex pass. As you suggested I will open an issue on mariaex. Actually I don't really use MariaDB,  but I wanted to check if this was the issue with postgrex or mariaex. Looking at the behaviour it seemed that it was the issue with Ecto because it acted the same for both database types, but now it is clear ;)

Btw, no problem for the app, I'm glad it helps you fix stuff faster!

I'm waiting for the new version of Postgrex then!

Regards
